### PR TITLE
Adds surgery: Lobotomies (Cake's request)

### PR DESCRIPTION
**Pull request by LordFowl**
As per Cake's looser instructions, MMI's will require a lobotomized brain to function. Instead of having the MMI automagically lobotomize a brain, I decided instead to make it a surgical operation and a prerequisite to borging. This also cleans up some snowflake istype() checks by instituting two new variables.

Cortical stacks, slime brains, and golem brains cannot be lobotomized, and as such cannot be put into an MMI.

You can lobotomize a person either by surgical-drilling their brain in surgery, or by surgical-drilling their removed brain.

The roboticist now spawns with a drill to accomodate this change. Surgical drills are not really useful otherwise, so I don't think this is a huge upset of the surgery balance.

Lobotomy will send the following message to the lobotomee;
"You have been lobotomized. Your memories and your former life have been surgically removed from your brain, and while you are lobotomized you remember nothing that ever came before this moment."

~~I personally believe this method to be superior to the idea of the MMI auto-lobotomizing, or chemically suppressing the brain, because it is more interactive and can be used without MMIs at all - I can see people performing lobotomies for various other purposes beyond MMIs. However, since there have been recently some complaints about the "realism" of a lobotomy, I will leave it to further discussion with Cake. Marking Do not merge until he has approved this.
These discussions have been hammered out. Merely leaving DNM until Cake confirms the changes,~~

As seems to be the trend with features specifically requested by lore-developers, this bypasses develop.

### CODE REVIEW

**Comment by CakeIsOssim** [comment]
I had a disagreement with the fluff text that appears after the lobotomy, but after some thought, I decided that it works as it is. Looks good to go.

**Comment by LordFowl** [comment]
Removing the DNM label, then. Opening it up for codereview, I suppose.

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 1]
```diff
@@ -34,29 +34,32 @@
...
+				user << "<span class='warning'>You aren't sure where this brain came from, but you're pretty sure it's a useless brain.</span>"
 				return
 
-			for(var/mob/V in viewers(src, null))
-				V.show_message(text("\blue [user] sticks \a [O] into \the [src]."))
+			user.visible_message("<span class='notice'>[user] sticks \a [B] into \the [src].</span>")
 
-			brainmob = O:brainmob
-			O:brainmob = null
+			brainmob = B:brainmob
```
Comment (skull132) [inline code/modules/mob/living/carbon/brain/MMI.dm]: Are the `:` operators necessary now? It might be able to compile with just `.`.

Comment (LordFowl) [inline code/modules/mob/living/carbon/brain/MMI.dm]: Old code that I neglected to fix.

**Pull request merged**